proj_eclipse/src/main/java/smartmag/models/OrderModel.java

```diff
@@ -147,11 +166,15 @@ public void setStatoInSvolgimento() {
 	}
 
 	/**
-	 * cancella il record dell'ordine nel modello
+	 * cancella il record dell'ordine nel modello poichè nel DB è stato messo
+	 * DELETE ON CASCADE, cancella anche i record id prodotti ordini
 	 */
-	public void deleteOrdine() throws ParseException {
+	@SuppressWarnings("unlikely-arg-type")
+	public void deleteRecords() throws ParseException {
 		if (orderIsSavedInDb()) {
 			orderRecord.delete(); // DELETE con UpdatableRecord
+			instances.remove(getOrderModelOf(ordine));
```

Comment:
di nuovo, ci limitiamo ad eliminare record e notificare; l'istanza del modello rimane
```suggestion
			orderRecord = null;
```